src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs

```diff
@@ -313,5 +321,159 @@ public static bool TryParse(string value, out SemanticVersion result)
         {
             return !(b < a);
         }
+
+        // private methods
+        private bool IsServerVersion(string preRelease)
+        {
+            if (preRelease == null)
+            {
+                return false;
+            }
+
+            var pattern = @"^((?<releaseCandidate>rc\d+)?-?(?<internalBuild>\d+-g[0-9a-f]{4,40})?)$";
+            var match = Regex.Match(preRelease, pattern);
+
+            return match.Groups["releaseCandidate"].Success || match.Groups["internalBuild"].Success;
```

Comment:
Lines 333-336 can be simplified to:

    var internalBuildPattern = @"^(rc\d+-)?\d+-g[0-9a-f]{4,40}$";
    return Regex.IsMatch(preRelease, internalBuildPattern);

Reply:
Not sure why we have `{4,40}`... does the number of characters in the commit hash vary?

Reply:
Generally speaking yes. The default is `7` which we currently use, but I would rather relax the regex in a non critical place in case this changes sometime.